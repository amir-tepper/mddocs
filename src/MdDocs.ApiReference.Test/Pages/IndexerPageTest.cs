using System;
using System.Linq;
using Grynwald.MdDocs.ApiReference.Configuration;
using Grynwald.MdDocs.ApiReference.Model;
using Grynwald.MdDocs.ApiReference.Model.XmlDocs;
using Grynwald.MdDocs.ApiReference.Pages;
using Microsoft.Extensions.Logging.Abstractions;

namespace Grynwald.MdDocs.ApiReference.Test.Pages
{
    /// <summary>
    /// Tests for <see cref="IndexerPage" />
    /// </summary>
    public class IndexerPageTest : PageTestBase<IndexerDocumentation, IndexerPage>
    {
        protected override IndexerPage CreatePage(IndexerDocumentation model, ApiReferenceConfiguration configuration)
        {
            return new IndexerPage(NullLinkProvider.Instance, configuration, model, NullLogger.Instance);
        }

        protected override IndexerDocumentation CreateSampleModel()
        {
            var assembly = Compile(@"
                namespace MyNamespace
                {
                    public class Class1
                    {
                        public int this[int i] { get { return 0; } }
                    }
                }
            ");

            var assemblyDocumentation = new AssemblyDocumentation(assembly, NullXmlDocsProvider.Instance, NullLogger.Instance);
            return assemblyDocumentation.MainModuleDocumentation.Types.Single().Indexers.Single();
        }
    }
}
