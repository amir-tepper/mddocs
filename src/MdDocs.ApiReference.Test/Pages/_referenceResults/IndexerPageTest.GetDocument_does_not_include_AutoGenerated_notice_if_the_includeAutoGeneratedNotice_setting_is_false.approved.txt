# Class1.Item Indexer

**Declaring Type:** Class1

```csharp
public int this[int i] { get; }
```

## Parameters

`i`  int

## Indexer Value

int

___

*Documentation generated by [MdDocs](https://github.com/ap0llo/mddocs)*
